metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/DropWizardGaugeRegistry.java

```diff
@@ -36,7 +41,16 @@ public DropWizardGaugeRegistry(MetricRegistry metricRegistry) {
 
     @Override
     public <T> GaugeRegistry register(String name, Gauge<T> gauge) {
-        metricRegistry.gauge(name, () -> gauge::get);
+        metricRegistry.gauge(name, () -> toCachedGauge(gauge));
         return this;
     }
+
+    private <T> CachedGauge<T> toCachedGauge(Gauge<T> gauge) {
```

Comment:
I don't understand what we want to achieve here.
Don't we want real value when we export a metric ?
If we compute them to often, is there anything we can do to tune the rate instead of cheating with a cache ?

Reply:
The goal is to circumvent an issue with es-reporter making it reading the gauge twice (first is a null check, second is a jackson serialization).
Maybe we should better find a way to fix the 2 reads.